core/io/config_file.cpp

```diff
@@ -183,7 +183,9 @@ Error ConfigFile::_internal_save(FileAccess *file) {
 		if (E != values.front()) {
 			file->store_string("\n");
 		}
-		file->store_string("[" + E.key() + "]\n\n");
+		if (E.key() != "" || E != values.front()) {
```

Comment:
It might be worth adding a comment to explain why this is done:

```suggestion
		// Store section-less keys without any section header.
		if (E.key() != "" || E != values.front()) {
```

Reply:
@Calinou Done.